Added support for Docker

AlexandrosAlexiou opened the pull request:
- Updated SetUp.md with a "Run locally with Docker" section
- Moved requirements.txt to the root of the repo since some hosting
  services require the file to be there (eg. [Heroku](https://elements.heroku.com/buildpacks/heroku/heroku-buildpack-python) )

AlexandrosAlexiou committed:
Added support for Docker

- Updated SetUp.md with a "Run locally with Docker" section
- Moved requirements.txt to the root of the repo since some hosting
  services require the file to be there (eg. Heroku https://elements.heroku.com/buildpacks/heroku/heroku-buildpack-python)

novatorem commented:
A couple of suggestions popped up from CodeFactor:
https://www.codefactor.io/repository/github/novatorem/novatorem/pull/44

I doubt there would be any problems from them anytime within the next couple of years, but thoughts?

AlexandrosAlexiou committed:
Apply codefactor.io suggestions

- Use specific python docker image (less error prone)
- Use '--no-cache-dir' flag in pip since pip cache makes docker images larger

AlexandrosAlexiou commented:
> A couple of suggestions popped up from CodeFactor: https://www.codefactor.io/repository/github/novatorem/novatorem/pull/44
> 
> I doubt there would be any problems from them anytime within the next couple of years, but thoughts?

- Using a specific python image is correct, since for example if we always use the latest python image we might have support errors with Flask (i think this is rare but still). I am running this with python 3.10.0 so I changed the latest keyword to 3.10.0.
- The pip cache indeed increases the docker image size. We have very few dependencies so caching would not be a problem, but nontheless I will comply with codefactor suggestions.

`I pushed a commit that applies codefactor.io suggestions.`
